front-end/src/views/function/components/AddFunc.vue

```diff
@@ -0,0 +1,242 @@
+<template>
+  <el-drawer
+    ref="addFuncDrawer"
+    size="40%"
+    title="AddFunction"
+  >
+    <el-form ref="addForm"
+             class="addFunc"
+             :rules="rules"
```

Comment:
Codacy has a fix for the issue: [Replace `·············` with `······`](https://app.codacy.com/gh/FunctionStream/function-stream/pullRequest?prid=7834759)
```suggestion
      :rules="rules"
```